### PR TITLE
Causal test case refactor

### PR DESCRIPTION
**Pull request by cwild-UoS**
This PR should simplify the structure of the CausalTestCase by assuming that there will only be a single treatment and control value per test case.

### CODE REVIEW

**Comment by cwild-UoS** [comment]
Branched off from https://github.com/CITCOM-project/CausalTestingFramework/pull/103

File: causal_testing/testing/causal_test_suite.py [hunk 1]
```diff
@@ -13,15 +13,15 @@ class CausalTestSuite(UserDict):
     base_test_case's and execute each causal_test_case with each iterator.
     """
 
-    def add_test_object(self, base_test_case, causal_test_case_list, estimators, estimate_type: str = "ate"):
+    def add_test_object(self, base_test_case, causal_test_case_list, estimators_classes, estimate_type: str = "ate"):
```
Comment (bobturneruk) [inline causal_testing/testing/causal_test_suite.py:16]: Should `estimators_classes` have a type in the header? I guess it's type would be `type`. This all ties in with the ABC stuff which I get a little lost in sometimes.

File: causal_testing/testing/estimators.py [hunk 2]
```diff
@@ -282,17 +282,17 @@ class LinearRegressionEstimator(Estimator):
...
     def __init__(
         self,
-        treatment: tuple,
-        treatment_values: float,
-        control_values: float,
-        adjustment_set: list[float],
+        treatment: tuple,    
+        treatment_value: float,
+        control_value: float,
+        adjustment_set: set,
```
Comment (bobturneruk) [inline causal_testing/testing/estimators.py:288]: This is another change that has had impacts elsewhere. Seems like it enables some quite neat code, but I guess we've lost control of the types that go in the set. `set[float]`?

**Review by bobturneruk — APPROVED** [review]
Looks like much of this stems from causal_testing/testing/causal_test_case.py line 40/41. Seems to all make sense and be neater code. Good stuff.

**Comment by bobturneruk** [comment]
Might be good for someone else in the team to get a quick glance at this from a usability perspective.

I suggest merging `causal_test_case_refactor` into `base-causal-test-case` once you're happy. Then we'll merge `base-causal-test-case` into `main`.

**Comment by jmafoster1** [comment]
This all looks good to me.

**Comment by cwild-UoS** [comment]
Cheers guys! I'll go through the feedback and make changes and do the merge tomorrow :).

File: causal_testing/testing/causal_test_case.py [hunk 3]
```diff
@@ -26,50 +30,46 @@ def __init__(
...
+        :param base_test_case: A BaseTestCase object consisting of a treatment variable, outcome variable and effect
+        :param expected_causal_effect: The expected causal effect (Positive, Negative, No Effect).
+        :param control_value: The control value for the treatment variable (before intervention).
+        :param treatment_value: The treatment value for the treatment variable (after intervention).
+        :param estimate_type: A string which denotes the type of estimate to return
+        :param effect_modifier_configuration:
         """
         self.base_test_case = base_test_case
-        self.control_input_configuration = {base_test_case.treatment_variable: control_value}
+        self.control_value = control_value
```
[the remark this answers is not in the document]
Comment (cwild-UoS) [reply causal_testing/testing/causal_test_case.py:41]: Yeah this is the root of the refactor. Converting the previous single key:value dictionaries into 2 separate attributes

File: causal_testing/testing/causal_test_suite.py (hunk 1 above)
Comment (cwild-UoS) [reply causal_testing/testing/causal_test_suite.py:16]: `Iterable[Type[Estimator]]`

I've chosen the above as the type hint as it should be a list or any iterable of casses that inherit from the `Estimator` class. 

That type hint helps me understand the datatype, but it's hard to say if there is a more technically accurate type hint to do with typing ABCs?

Comment (bobturneruk) [reply causal_testing/testing/causal_test_suite.py:16]: Not that I'm aware of.

File: causal_testing/testing/estimators.py (hunk 2 above)
Comment (cwild-UoS) [reply causal_testing/testing/estimators.py:288]: The change in typing for `adjustment_set` was to make it consistent with the typing of `adjustment_set` in the other estimator classes.

**Pull request merged**